ocean_provider/routes/consume.py

```diff
@@ -276,8 +277,10 @@ def download():
         )
 
         file_index = int(data.get("fileIndex"))
-
-        url_object = get_service_files_list(service, provider_wallet)[file_index]
+        files_list = get_service_files_list(service, provider_wallet)
+        if file_index > len(files_list):
+            return jsonify(error=f"No such fileIndex")
```

Comment:
can you also add a 400 status code? And I would suggest a tighter condition on the if, e.g. if the file index is not found in the file list. Possibly it's `if file_index not in files_list` or `if file_index not in enum(files_list)`

Reply:
agreed with 400.   but for tighter condition, it's enough.  files_list is an array, you cannot have files_list[0] and files_list[2] , but file_.list[1] is missing.